### ISSUE TITLE
fine-tune/evaluation on sen12ms

### ISSUE THREAD

**Issue by taeil** [Berkeley-Data/hpt#44]
Yeah - Sen12ms may not be the perfect evaluation because of the mislableling issue, but it's a good diagnostic to make sure our system is working properly -- and we can report the results either way.
The initial tricky part with using an RGB dataset (like RESISC) for evaluation is that the 3 channel input is not expecting RGB, but rather, the output of our input_modules (I think the right way to do this is to perform a bit of moco training on the target dataset where we freeze all params except an appropriate input_module -- but this will take a little engineering to set up).
So for now, it'll be easier if we use Sen1 and Sen2 for evaluation, and use their corresponding input_module that trained during moco pretraining.

A 200 epoch model is at:
/scratch/crguest/vivid-resonance-73_sen12ms_no_aug_200epoch.pth
where the wandb training is here: https://wandb.ai/cjrd/BDOpenSelfSup-tools/runs/3qjvxo2p?workspace=user-cjrd

**Comment by taeil** [comment]
[Option 1: SEN12MS](https://github.com/Berkeley-Data/SEN12MS/tree/taeil#wip):
modifying SEN12MS/classification/main_train.py for fine_tuning. 

Stuck at creating the model with same architecture.  😞 

[Option 2: OpenSelfSup]

**Comment by taeil** [comment]
https://github.com/Berkeley-Data/hpt/issues/48

**Comment by taeil** [comment]
[wandb dashboard](https://wandb.ai/cal-capstone/SEN12MS/table)

**Comment by taeil** [comment]
Keeping everything the same except the pre-trained model, how about we run finetune on 2000 labels and evaluate?  
does the Moco pre-trained model show better comparing to supervised training?

**Comment by suryatechie** [comment]
Good idea @taeil . We can give it a try. Since the SEN12MS data has noise in terms of labeling, let us try using 2000 labels.

**Comment by taeil** [comment]
> Good idea @taeil . We can give it a try. Since the SEN12MS data has noise in terms of labeling, let us try using 2000 labels.

okie, I will train 4 models (RGB, s1, s2, s1 & s2) with baseline training.

**Comment by taeil** [comment]
should run on 2000 datasets since the result is not super strong

**Comment by taeil** [comment]
updated to run finetune and eval at the same time.

**Comment by taeil** [comment]
Need to adjust the number of layers to train to see which one may have better performance. Ideally, only top layers. 
- https://cs231n.github.io/transfer-learning/ 

> New dataset is small and similar to original dataset. Since the data is small, it is not a good idea to fine-tune the ConvNet due to overfitting concerns. Since the data is similar to the original data, we expect higher-level features in the ConvNet to be relevant to this dataset as well. Hence, the best idea might be to train a linear classifier on the CNN codes.
